src/uu/chmod/src/chmod.rs

```diff
@@ -37,13 +37,54 @@ mod options {
 
 #[uucore::main]
 pub fn uumain(args: impl uucore::Args) -> UResult<()> {
-    let mut args = args.collect_lossy();
+    let args = args.collect_lossy();
 
-    // Before we can parse 'args' with clap (and previously getopts),
-    // a possible MODE prefix '-' needs to be removed (e.g. "chmod -x FILE").
-    let mode_had_minus_prefix = mode::strip_minus_from_mode(&mut args);
+    // "-w -w" is the same as "-w,-w"
+    // "-w file -w" is the same as "-w,-w file"
+    // "-- -w file" is the same as "-w file"
+    let mut parsed_cmode_vec = Vec::new();
+    let double_hyphen = args.iter().find(|&x| x == "--");
+    let mut args_iter = args.iter();
+    let mut clean_args: Vec<&String> = args_iter
```

Comment:
A slightly different flow might be possible here to reduce the number of vecs. You can start with the binary name to get the `args_vec` immediately (assuming the pseudo cmode can be removed):
```rust
let mut clean_args = vec![args_iter.next()];
clean_args.extend(
    args_iter
        .by_ref()
        .take_while(...)
        .filter(...)
)
```